components/Match/analysis.js

```diff
@@ -0,0 +1,341 @@
+        category: 'primary',
+        icon: 'fa-battery-4',
+        valid: runes !== undefined && pm.lane_role === 2,
+        score(raw)
+        {
+          return raw / target;
+        },
+        top: target,
+      };
+    },
+    // unused item actives (multiple results?)
+    unused_item(m, pm)
+    {
+      const result = [];
+      if (pm.purchase)
+      {
+        for (const key in pm.purchase)
+        {
+          if (pm.purchase[key] && getGroupedItemUses(key) < 1 && constants.items[key] && isActiveItem(key))
+          {
+            // if item has cooldown, consider it usable
+            result.push("<img title='" + key + "' class='item img-sm' src='" + constants.items[key].img + "' />");
+          }
+        }
+      }
+
+      function getGroupedItemUses(key)
+      {
+        let total = 0;
+        for (const key2 in pm.item_uses)
+        {
+          if (key === key2 || constants.item_groups.some(function (g)
+            {
+            return (key in g) && (key2 in g);
+          }))
+          {
+            total += pm.item_uses[key];
+          }
+        }
+        return total;
+      }
+      return {
+        abbr: 'ITEMUSE',
+        name: 'Unused active items',
+        suffix: util.format('%s', result.length ? result.join('') : 0),
+        value: result.length,
+        advice: 'Make sure to use your item actives in order to fully utilize your investment.',
+        category: 'success',
+        icon: 'fa-bolt',
+        valid: pm.purchase,
+        score(raw)
+        {
+          return 5 - raw;
+        },
+        top: 0,
+      };
+    },
+  };
+  for (const key in checks)
+  {
+    advice[key] = checks[key](match, pm);
+    const val = advice[key];
+    val.display = util.format('%s: <b>%s</b>, expected <b>%s</b>', val.name, Number(val.value ? val.value.toFixed(2) : ''), Number(val.top.toFixed(2)));
+    val.display += (val.suffix ? ' ' + val.suffix : '');
+    val.pct = val.score(val.value) / val.score(val.top);
+    delete val.score;
+    pm.desc = [constants.lane_role[pm.lane_role], isSupport(pm) ? 'Support' : 'Core'].join('/');
+  }
+  return advice;
+
+  function isSupport(pm)
+  {
+    return getObsWardsPlaced(pm) >= 2 && pm.lh_t && pm.lh_t[10] < 20;
+  }
+
+  function getObsWardsPlaced(pm)
+  {
+    if (!pm.obs_log)
+    {
+      return 0;
+    }
+    return pm.obs_log.filter(function (l)
+    {
+      return !l.entityleft;
+    }).length;
+  }
+
+  function isRoshHero(pm)
+  {
+    const rosh_heroes = {
+      'npc_dota_hero_lycan': 1,
+      'npc_dota_hero_ursa': 1,
+      'npc_dota_hero_troll_warlord': 1,
+    };
+    return constants.heroes[pm.hero_id] && (constants.heroes[pm.hero_id].name in rosh_heroes);
+  }
+
+  function isActiveItem(key)
+  {
+    const whitelist = {
+      'branches': 1,
+      'bloodstone': 1,
+      'radiance': 1,
+    };
+    return (constants.items[key].desc.indexOf('Active: ') > -1 && !(key in whitelist));
+  }
+}
+export
+  {
+    generatePlayerAnalysis,
+  };
```

Comment:
I think we should convert this to our lint standards sooner rather than later, but if it's a direct port it's not necessary asap if you think time is better spent updating functionality. But we should keep an issue open so we don't lose track of it.